### PR TITLE
Tls sipcert per acc

### PR DESCRIPTION
**Pull request by cHuberCoffee**
This PR depends on [#96](https://github.com/baresip/re/pull/96)

It allows the user to specify different client certificates for different
accounts in the accounts file.

SIP proxies environments may not allow installing new trusted certificates. In 
the same time they may have a certificate installed which is used to issue
new SIP certificates. For a single server it is no problem, since a user could
configure the server generated certificate as the SIP_certificate in the
config file. Having multiple such servers (maybe as fallback servers)
leads to a problem where baresip is currently not able to handle a variety of
client certificates.

This PR overcomes exactly the described problem.
In case a user has more than one account on the same SIP proxy. The connection
TLS handshake is done once. Therefore the client certificate check of the
server is done once. Configuring different client certificates for the same 
server configuration makes no sense and should be avoided.

Since the certificate option in the accounts file is optional, baresip uses
the SIP_certificate of the config file as fallback.

### CODE REVIEW

File: docs/examples/accounts [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #    ;auth_user=username
 #    ;auth_pass=password
 #    ;call_transfer=no
+#    ;cert=cert.pem
```
Comment (alfredh) [inline docs/examples/accounts:18]: what kind of certificate is this ?

is it Client cert, or Server cert ?

File: src/ua.c [hunk 2]
```diff
@@ -977,6 +977,7 @@ int ua_alloc(struct ua **uap, const char *aor)
 	if (err)
 		goto out;
 
+
```
Comment (alfredh) [inline src/ua.c:980]: this is an unrelated change ...

File: src/ua.c [hunk 3]
```diff
@@ -1000,6 +1001,18 @@ int ua_alloc(struct ua **uap, const char *aor)
 			  ua->acc->menc->id);
 	}
 
+	if (ua->acc->cert) {
+		err = sip_transp_add_ccert(uag.sip,
```
Comment (alfredh) [inline src/ua.c:1005]: does ccert mean client certificate ?

**Comment by alfredh** [comment]
if this patch adds support for per-account certificates, should we then consider to
remove support for global certificate ?

having only one way to do things makes it easier to use and reduces the codesize.

**Comment by juha-h** [comment]
Alfred E. Heggestad writes:

> if this patch adds support for per-account certificates, should we
> then consider to remove support for global certificate ?

I would prefer to keep the global one for backwards compatibility.
Also, it can be considered as the device certificate.

May be config certificate could be considered as an account certificate
for those accounts that don't have one.

**Comment by alfredh** [comment]
@sreimers can you please merge the re patch ?

**Comment by sreimers** [comment]
> @sreimers can you please merge the re patch ?

Done
